matrix-event.c

```diff
@@ -50,5 +50,6 @@ void matrix_event_free(MatrixRoomEvent *event)
     g_free(event->txn_id);
     g_free(event->sender);
     g_free(event->event_type);
+    g_free(event->hook_data);
```

Comment:
I've added a comment about the void *hook_data;    my main reasons for doing it here was that matrix_event_free frees pretty much everything else on the event, and also I don't have to worry about thinking about the life time of the event; if something decides it wants to flush all the events then it can; although there's then a question of whether something else should happen (for example unref the image in the purple image store); but how would we do that? Perhaps we need a hook callback with a flag?